test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -4167,6 +4169,148 @@ TEST_P(SslSocketTest, RevokedCertificateCRLInTrustedCA) {
   testUtil(successful_test_options.setExpectedSerialNumber(TEST_SAN_DNS2_CERT_SERIAL));
 }
 
+TEST_P(SslSocketTest, RevokedIntermediateCertificate) {
+
+  // This should succeed, since the crl chain is complete.
+  const std::string complete_server_ctx_yaml = R"EOF(
+  common_tls_context:
+    tls_certificates:
+      certificate_chain:
+        filename: "{{ test_tmpdir }}/unittestcert.pem"
+      private_key:
+        filename: "{{ test_tmpdir }}/unittestkey.pem"
+    validation_context:
+      trusted_ca:
+        filename: "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/intermediate_ca_cert_chain.pem"
+      crl:
+        filename: "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/intermediate_ca_cert.crl"
+)EOF";
+
+  // This should fail, since the crl chain is incomplete.
+  const std::string incomplete_server_ctx_yaml = R"EOF(
+  common_tls_context:
+    tls_certificates:
+      certificate_chain:
+        filename: "{{ test_tmpdir }}/unittestcert.pem"
+      private_key:
+        filename: "{{ test_tmpdir }}/unittestkey.pem"
+    validation_context:
+      trusted_ca:
+        filename: "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/intermediate_ca_cert_chain.pem"
+      crl:
+        filename: "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/intermediate_ca_cert_chain.crl"
```

Comment:
Please add a comment saying it is in the different crl files.

Reply:
Updated with comment within the YAML snippets.  Let me know if this is what you were thinking.

Reply:
Unfortunately I was using the branch to move code from a development machine to a test machine and was being a bit cavalier about things which I now regret myself.  I'll maintain individual commits moving forward to address code reviews.

Regarding the format failures, those mostly come from #12755 which stands on its own to fix the test certificate generation bit that's broken as per #12752.  I think I'd like that merged in then I can just rebase (or rather merge in) the changes to get this thing on the right foot.

If you'd rather I just do it all here I'm okay with that too but the two are only somewhat related in that this should build on that.